packages/gatsby-theme-store/src/components/Error/ErrorHandler.tsx

```diff
@@ -29,18 +27,21 @@ export const handleError = ({ error, errorId }: Props) => {
     return
   }
 
-  if (isUserOffline) {
-    const previousPagePath = encodeURIComponent(window.location.pathname)
+  const from = encodeURIComponent(window.location.pathname)
 
-    window.location.href = `/offline?from=${previousPagePath}`
+  const isOffline = !window.navigator.onLine
+  const is404 = error?.extensions?.exception?.status === 404
+  const isFrameworkLevelError = error.name === 'ChunkLoadError'
 
-    return
+  if (isFrameworkLevelError) {
+    window.location.reload()
+  } else if (isOffline) {
+    window.location.href = `/offline?from=${from}`
+  } else if (is404) {
+    window.location.href = `/404?from=${from}`
+  } else {
+    window.location.href = `/500?from=${from}&errorId=${errorId ?? uuidv4()}`
```

Comment:
we need to add sentry here, in an agnostic way to the store can configure a own sentry

Reply:
This was like this before, but I removed this if because sometimes we do want to see the error handling in action during development. Also, this ensure the same behavior in dev and in prod

Reply:
We do have a way for adding sentry: https://github.com/vtex-sites/marinbrasil.store/blob/master/src/%40vtex/gatsby-theme-store/components/Error/ErrorHandler.tsx

Reply:
yes, but we only generate a random id with uuid when the store does not shadowing this file.

Reply:
Yes. AFAIK sentry is open source right?